core/embed/lib/fonts/font_tthoves_bold_upper_17.c

```diff
@@ -0,0 +1,205 @@
+/* Y */ static const uint8_t Font_TTHoves_Bold_upper_17_glyph_89[] = { 13, 12, 12, 0, 12, 225, 255, 1, 0, 242, 223, 0, 112, 255, 8, 0, 250, 95, 0, 0, 254, 31, 32, 255, 12, 0, 0, 246, 143, 160, 255, 4, 0, 0, 224, 255, 243, 207, 0, 0, 0, 96, 255, 255, 63, 0, 0, 0, 0, 253, 255, 11, 0, 0, 0, 0, 245, 255, 3, 0, 0, 0, 0, 224, 223, 0, 0, 0, 0, 0, 224, 207, 0, 0, 0, 0, 0, 224, 207, 0, 0, 0, 0, 0, 224, 207, 0, 0, 0 };
+/* Z */ static const uint8_t Font_TTHoves_Bold_upper_17_glyph_90[] = { 10, 12, 11, 0, 12, 243, 255, 255, 255, 207, 243, 255, 255, 255, 207, 162, 170, 170, 255, 175, 0, 0, 144, 255, 30, 0, 0, 245, 255, 3, 0, 32, 255, 111, 0, 0, 209, 255, 9, 0, 0, 251, 207, 0, 0, 128, 255, 30, 0, 0, 243, 255, 156, 153, 137, 245, 255, 255, 255, 223, 245, 255, 255, 255, 223 };
+/* [ */ static const uint8_t Font_TTHoves_Bold_upper_17_glyph_91[] = { 5, 17, 7, 1, 13, 83, 85, 5, 250, 255, 14, 250, 255, 14, 250, 14, 0, 250, 14, 0, 250, 14, 0, 250, 14, 0, 250, 14, 0, 250, 14, 0, 250, 14, 0, 250, 14, 0, 250, 14, 0, 250, 14, 0, 250, 14, 0, 250, 95, 5, 250, 255, 14, 250, 255, 14 };
+/* \ */ static const uint8_t Font_TTHoves_Bold_upper_17_glyph_92[] = { 8, 16, 7, 0, 12, 253, 11, 0, 0, 249, 15, 0, 0, 244, 79, 0, 0, 240, 159, 0, 0, 176, 223, 0, 0, 96, 255, 2, 0, 16, 255, 7, 0, 0, 252, 11, 0, 0, 248, 15, 0, 0, 243, 95, 0, 0, 224, 159, 0, 0, 160, 239, 0, 0, 80, 255, 3, 0, 16, 255, 7, 0, 0, 252, 12, 0, 0, 247, 31 };
+/* ] */ static const uint8_t Font_TTHoves_Bold_upper_17_glyph_93[] = { 6, 17, 7, 0, 13, 82, 85, 21, 246, 255, 47, 246, 255, 47, 0, 245, 47, 0, 245, 47, 0, 245, 47, 0, 245, 47, 0, 245, 47, 0, 245, 47, 0, 245, 47, 0, 245, 47, 0, 245, 47, 0, 245, 47, 0, 245, 47, 82, 249, 47, 246, 255, 47, 246, 255, 47 };
+/* ^ */ static const uint8_t Font_TTHoves_Bold_upper_17_glyph_94[] = { 8, 5, 8, 0, 12, 0, 251, 159, 0, 16, 255, 255, 0, 112, 207, 253, 5, 192, 111, 248, 10, 242, 31, 242, 31 };
+/* _ */ static const uint8_t Font_TTHoves_Bold_upper_17_glyph_95[] = { 9, 3, 8, 0, 0, 82, 85, 85, 85, 0, 245, 255, 255, 255, 1, 245, 255, 255, 255, 1 };
+/* ` */ static const uint8_t Font_TTHoves_Bold_upper_17_glyph_96[] = { 5, 3, 5, 0, 13, 114, 23, 0, 176, 175, 0, 0, 253, 5 };
+/* a -> A */ static const uint8_t Font_TTHoves_Bold_upper_17_glyph_97[] = { 13, 12, 13, 0, 12, 0, 0, 243, 255, 31, 0, 0, 0, 0, 248, 255, 111, 0, 0, 0, 0, 253, 255, 191, 0, 0, 0, 48, 255, 168, 255, 1, 0, 0, 128, 255, 83, 255, 6, 0, 0, 224, 239, 0, 255, 12, 0, 0, 243, 159, 0, 251, 31, 0, 0, 249, 255, 255, 255, 111, 0, 0, 254, 255, 255, 255, 207, 0, 48, 255, 107, 102, 198, 255, 1, 144, 255, 5, 0, 112, 255, 7, 224, 255, 0, 0, 32, 255, 12 };
```

Comment:
Why not just generate `static const uint8_t Font_TTHoves_Bold_upper_17_glyph_97 = Font_TTHoves_Bold_upper_17_glyph_65;`. Seems like easy win of couple KiB, especially in bootloader, but maybe I'm missing some downside.

Reply:
You are right, now here only upper case letters are generated and they are referenced in the font array below
e43624f2d6f9f80ed624123d9c30d9ea325b239a

This saves some space for model T. 
Regarding model R, it helps a bit but a lot of it is still duplicated, you are right.
But what is the memory situation there? How do I know?

Reply:
We can modify the generator further to link `_upper` fonts to the existing font though. :thinking:

Reply:
Is this job accurate? If it is, then we should be fine. https://github.com/trezor/trezor-firmware/actions/runs/8884320731/job/24393365753?pr=3696
![image](https://github.com/trezor/trezor-firmware/assets/14001709/37f3ee1f-b6c6-4682-835f-74d9f26f8842)

Reply:
`gen_font.py` modified here 8547def0d12d63b7de9910bb9450478fd650b78f 

Now, the glyphs are stored only once and the font arrays point to the same glyphs. Nonprintable glyph is also stored only once and `_upper` version is a pointer to the glyph. The result in `.h` files look like this:

```
extern const uint8_t* const Font_PixelOperator_Regular_8[126 + 1 - 32];
extern const uint8_t Font_PixelOperator_Regular_8_glyph_nonprintable[];

extern const uint8_t* const Font_PixelOperator_Regular_8_upper[126 + 1 - 32];
extern const uint8_t Font_PixelOperator_Regular_8_upper_glyph_nonprintable[];
```